File: erp_api/ERP.API/Features/Services/ServiceRequest.cs
namespace ERP.API.Features.Services;

public class ServiceRequest
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Priority { get; set; } = "Medium"; // Low, Medium, High, Critical
    public string Status { get; set; } = "Open"; // Open, InProgress, Resolved, Closed
    public string RequestedBy { get; set; } = string.Empty;
    public string AssignedTo { get; set; } = string.Empty;
    public string Department { get; set; } = string.Empty;
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
    public DateTime? ResolvedDate { get; set; }
}

File: erp_flutter/lib/core/models/service_request.dart
class ServiceRequest {
  final int id;
  final String title;
  final String description;
  final String priority;
  final String status;
  final String requestedBy;
  final String assignedTo;
  final String department;
  final DateTime createdDate;
  final DateTime? resolvedDate;

  ServiceRequest({
    required this.id,
    required this.title,
    required this.description,
    required this.priority,
    required this.status,
    required this.requestedBy,
    required this.assignedTo,
    required this.department,
    required this.createdDate,
    this.resolvedDate,
  });

  factory ServiceRequest.fromJson(Map<String, dynamic> json) {
    return ServiceRequest(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      priority: json['priority'] ?? '',
      status: json['status'] ?? '',
      requestedBy: json['requestedBy'] ?? '',
      assignedTo: json['assignedTo'] ?? '',
      department: json['department'] ?? '',
      createdDate: json['createdDate'] != null ? DateTime.parse(json['createdDate']) : DateTime.now(),
      resolvedDate: json['resolvedDate'] != null ? DateTime.parse(json['resolvedDate']) : null,
    );
  }
}
